### PR TITLE
Don't activate the error list when parsing error logs, only show it. …

### PR DESCRIPTION
**Pull request by GabeDeBacker**
This fix address issue #222 which was causing keyboard focus to be taken away from the user when an extension called the open log APIs.

…It is annoying to have keyboard focus yanked away from you while log files are processing in the background.

### CODE REVIEW

File: src/Sarif.Viewer.VisualStudio/SdkUiUtilities.cs [hunk 1]
```diff
@@ -388,6 +388,40 @@ public static bool TryGetActiveViewForTextBuffer(ITextBuffer textBuffer, out IWp
...
+        /// Locates a tool window and request that it be shown.
+        /// </summary>
+        /// <param name="toolWindowGuid">A tool window identifier, typically one from <see cref="ToolWindowGuids80"/> such as the error list.</param>
+        /// <param name="activate">Indicates whether to activate (take keyboard focus) when showing the tool window.</param>
+        /// <returns>Returns a task that when completed indicates if the tool window was shown.</returns>
+        public static async System.Threading.Tasks.Task<bool> ShowToolWindowAsync(Guid toolWindowGuid, bool activate)
+        {
+            await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
+            IVsUIShell uiShell = Microsoft.VisualStudio.Shell.Package.GetGlobalService(typeof(SVsUIShell)) as IVsUIShell;
+            if (uiShell == null)
```
Comment (jeffersonking) [inline src/Sarif.Viewer.VisualStudio/SdkUiUtilities.cs:401]: Use null-conditional below to skip this check?

File: src/Sarif.Viewer.VisualStudio/ErrorList/ErrorListService.cs [hunk 2]
```diff
@@ -215,12 +215,6 @@ public static async System.Threading.Tasks.Task ProcessLogFileAsync(string fileP
             }
 
             await ProcessSarifLogAsync(log, outputPath, showMessageOnNoResults: promptOnLogConversions, cleanErrors: cleanErrors).ConfigureAwait(continueOnCapturedContext: false);
-
-            await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
-            if (AsyncPackage.GetGlobalService(typeof(DTE)) is DTE2 dte)
-            {
-                dte.ExecuteCommand("View.ErrorList");
```
Comment (lgolding) [inline src/Sarif.Viewer.VisualStudio/ErrorList/ErrorListService.cs:222]: >ErrorList [](start = 41, length = 9)

So if we don't do this, when does the viewer see the error list? I understand you don't want to pull focus to it, but if (1) if it wasn't previously opened you do want to open it, and (2) if it's open but not the currently selected tab in -- in whatever you call that collection of windows at the bottom -- then you do want to make that the selected tab. IMO.

Comment (lgolding) [reply src/Sarif.Viewer.VisualStudio/ErrorList/ErrorListService.cs:222]: I see your PR description explicitly says "It's ok to "show" the Error list"... so I will keep looking for the code that does that.

---
In reply to: [504119046](https://github.com/microsoft/sarif-visualstudio-extension/pull/223#discussion_r504119046) [](ancestors = 504119046)

**Review by lgolding — APPROVED** [review]
:shipit:

File: src/Sarif.Viewer.VisualStudio/ErrorList/ErrorListService.cs [hunk 3]
```diff
@@ -381,9 +375,12 @@ internal static async System.Threading.Tasks.Task ProcessSarifLogAsync(SarifLog
...
             }
 
-            if (!hasResults && showMessageOnNoResults)
+            if (hasResults)
+            {
+                SdkUIUtilities.ShowToolWindowAsync(new Guid(ToolWindowGuids80.ErrorList), activate: false).FileAndForget("Microsoft/SARIF/Viewer/ShowErrorList");
+            }
+            else if (showMessageOnNoResults)
             {
-                await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
```
[the remark this answers is not in the document]
Comment (GabeDeBacker) [reply src/Sarif.Viewer.VisualStudio/ErrorList/ErrorListService.cs:386]: No, it is already being done at the top of this function.

---
In reply to: [504117780](https://github.com/microsoft/sarif-visualstudio-extension/pull/223#discussion_r504117780) [](ancestors = 504117780)

File: src/Sarif.Viewer.VisualStudio/SdkUiUtilities.cs (hunk 1 above)
Comment (GabeDeBacker) [reply src/Sarif.Viewer.VisualStudio/SdkUiUtilities.cs]: This was lifted right from VS's codebase from their utility class.

Cleaned it up.

---
In reply to: [504106375](https://github.com/microsoft/sarif-visualstudio-extension/pull/223#discussion_r504106375) [](ancestors = 504106375)

**Review by jeffersonking — APPROVED** [review]

**Pull request merged**